app.js

```diff
@@ -1,11 +1,125 @@
 /* Imports */
-
+import { renderDiv, renderPlayer } from './render-utils.js';
 /* Get DOM Elements */
-
+const goblinsDefeatedEl = document.getElementById('goblins-defeated');
+const nameInput = document.getElementById('name-input');
+const nameButton = document.getElementById('name-button');
+const goblinsEl = document.querySelector('.goblins-container');
+const playerStatsEl = document.querySelector('.player-div');
+const fighterImageEl = document.getElementById('fighter-image');
 /* State */
+let player = { HP: 10, attack: 11, defense: 4 };
+let goblinsDefeated = 0;
+let attacksLanded = 0;
+let totalAttacks = 0;
+let counterattacksLanded = 0;
+
+// Starter Goblins
+const goblins = [
+    { name: 'Drew', health: 4, attack: 8, defense: 2 },
+    { name: 'Korbin', health: 3, attack: 8, defense: 4 },
+    { name: 'Lisa', health: 4, attack: 10, defense: 3 },
+    { name: 'Dave', health: 2, attack: 9, defense: 3 },
+];
 
 /* Events */
+nameButton.addEventListener('click', () => {
+    const newGoblin = {
+        name: nameInput.value || `Goober #${Math.ceil(Math.random() * 100)}`,
+        health: Math.ceil(Math.random() * 7),
+        attack: 8 + Math.floor(Math.random() * 6),
+        defense: 2 + Math.floor(Math.random() * 2),
+    };
+    goblins.push(newGoblin);
+    nameInput.value = '';
+    displayGoblins();
+});
 
 /* Display Functions */
+function displayGoblins() {
```

Comment:
This is a MONSTER!

Reply:
agreed. I plan to trim it down by function-izing some stuff.